### PR TITLE
Add pyfunc soft_nms for eval.

### PR DESCRIPTION
**Pull request by mingxingtan**

### CODE REVIEW

**Comment by mingxingtan** [comment]
@fsx950223 soft_nms slightly improves COCO accuracy by about 0.3 or 0.4.

File: efficientdet/anchors.py [hunk 1]
```diff
@@ -139,7 +140,80 @@ def nms(dets, thresh):
...
+      weight[iou > iou_thresh] -= iou[iou > iou_thresh]
+    elif method == 'gaussian':
+      weight = np.exp(-(iou * iou) / sigma)
+    else:  # traditional nms
+      weight = np.ones_like(iou)
+      weight[iou > iou_thresh] = 0
+
+    dets[1:, 4] *= weight
+    retained_idx = np.where(dets[1:, 4] >= score_thresh)[0]
+    dets = dets[retained_idx + 1, :]
```
Comment (fsx950223) [inline efficientdet/anchors.py:214]: Same as nms?

File: efficientdet/anchors.py [hunk 2]
```diff
@@ -139,7 +140,80 @@ def nms(dets, thresh):
...
+    inter = w * h
+    iou = inter / (dets[0, 5] + dets[1:, 5] - inter)
+
+    if method == 'linear':
+      weight = np.ones_like(iou)
+      weight[iou > iou_thresh] -= iou[iou > iou_thresh]
+    elif method == 'gaussian':
+      weight = np.exp(-(iou * iou) / sigma)
+    else:  # traditional nms
+      weight = np.ones_like(iou)
```
Comment (fsx950223) [inline efficientdet/anchors.py:209]: Same as nms?

**Comment by fsx950223** [comment]
> @fsx950223 soft_nms slightly improves COCO accuracy by about 0.3 or 0.4.

What's config get the increment?

File: efficientdet/anchors.py (hunk 2 above)
Comment (mingxingtan) [reply efficientdet/anchors.py:209]: yes, the output is the same, but the other nms function runs slightly faster than this one (due to the implementation).

**Comment by mingxingtan** [comment]
Just set method=gaussian would bump D7 testdev accuracy to 53.7 (from the original 53.4), but I haven't verified other models.
I will update the README soon.

**Comment by mingxingtan** [comment]
I am going to merge this PR, and we can refine it later.

**Pull request merged**